### PR TITLE
JSDoc comments for JS

### PR DESCRIPTION
**Pull request by benitav**
Initial goal:
- [ ] Make sure common classes and methods have JSDoc style comments
    - [x] React Native SDK (N/A - mostly references sdk-common classes and methods)
    - [x] React SDK (N/A - hooks already documented)
    - [x] Common SDK
    - [ ] Attachments
    - [ ] Web SDK
- [x] Figure out how to get [references] to work inside JSDoc comments
- [x] Before merging, resolve new "TODO" references

### CODE REVIEW

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts [hunk 1]
```diff
@@ -466,6 +488,12 @@ export abstract class AbstractPowerSyncDatabase extends BaseObserver<PowerSyncDB
     });
   }
 
+  /**
+   * Open a read-only transaction.
+   * TODO: Up to maxReaders read transactions can run concurrently. After that, read transactions are queued.
```
Comment (benitav) [inline packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:493]: What's the equivalent of maxReaders in sdk-common (if exists)?

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts [hunk 2]
```diff
@@ -481,6 +509,11 @@ export abstract class AbstractPowerSyncDatabase extends BaseObserver<PowerSyncDB
     );
   }
 
+  /**
+   * Open a read-write transaction.
+   * This takes a global lock - only one write transaction can execute against the database at a time.
+   * TODO: Statements within the transaction must be done on the provided SqliteWriteContext - attempting statements on the SqliteConnection instance will error.
```
Comment (benitav) [inline packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:515]: Is this relevant to this JS implementation? If so, what is the equivalent of SqliteWriteContext and SqliteConnection (these are from Flutter).

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts [hunk 3]
```diff
@@ -496,6 +529,11 @@ export abstract class AbstractPowerSyncDatabase extends BaseObserver<PowerSyncDB
     );
   }
 
+  /**
+   * Execute a read query every time the source tables are modified.
+   * TODO: Use throttle to specify the minimum interval between queries.
```
Comment (benitav) [inline packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:534]: I couldn't see a throttle property in this implementation - please confirm if sdk-common has an equivalent.

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts [hunk 4]
```diff
@@ -496,6 +529,11 @@ export abstract class AbstractPowerSyncDatabase extends BaseObserver<PowerSyncDB
...
+   * TODO: Source tables are automatically detected using `EXPLAIN QUERY PLAN`.
```
Comment (benitav) [inline packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:535]: Same as above, please confirm if this is applicable here and what the equivalent is.

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts [hunk 5]
```diff
@@ -582,6 +620,9 @@ export abstract class AbstractPowerSyncDatabase extends BaseObserver<PowerSyncDB
     });
   }
 
+  /**
+   * TODO
```
Comment (benitav) [inline packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:624]: Couldn't find an equivalent on the [Flutter side](https://pub.dev/documentation/powersync/latest/powersync/PowerSyncDatabase-class.html) - needs a description.

File: packages/powersync-sdk-common/src/client/sync/bucket/CrudBatch.ts [hunk 6]
```diff
@@ -1,9 +1,21 @@
 import { CrudEntry } from './CrudEntry';
 
+/**
+ * TODO
```
Comment (benitav) [inline packages/powersync-sdk-common/src/client/sync/bucket/CrudBatch.ts:4]: Needs a description.

File: packages/powersync-sdk-common/src/client/sync/bucket/CrudEntry.ts [hunk 7]
```diff
@@ -67,6 +88,9 @@ export class CrudEntry {
     this.transactionId = transactionId;
   }
 
+  /**
+   * TODO: Converts the change to JSON format, as required by the dev crud API.
```
Comment (benitav) [inline packages/powersync-sdk-common/src/client/sync/bucket/CrudEntry.ts:92]: Does sdk-common also have a "dev crud API" (assume it's referring to [this](https://pub.dev/documentation/powersync/latest/powersync/DevConnector-class.html))

File: packages/powersync-sdk-common/src/client/sync/bucket/CrudTransaction.ts [hunk 8]
```diff
@@ -1,10 +1,22 @@
 import { CrudBatch } from './CrudBatch';
 import { CrudEntry } from './CrudEntry';
 
+/**
+ * TODO
```
Comment (benitav) [inline packages/powersync-sdk-common/src/client/sync/bucket/CrudTransaction.ts:5]: Needs a description

File: packages/powersync-sdk-common/src/db/crud/SyncStatus.ts [hunk 9]
```diff
@@ -14,14 +14,23 @@ export type SyncStatusOptions = {
...
 
+  /**
+   * true if currently connected.
+   */
   get connected() {
     return this.options.connected ?? false;
   }
 
+  /**
+   * TODO
```
Comment (benitav) [inline packages/powersync-sdk-common/src/db/crud/SyncStatus.ts:25]: Needs a description

File: packages/powersync-sdk-common/src/client/sync/bucket/CrudEntry.ts (hunk 7 above)
Comment (benitav) [inline packages/powersync-sdk-common/src/client/sync/bucket/CrudEntry.ts:92]:
```suggestion
   * Converts the change to JSON format.
```

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts (hunk 3 above)
Comment (benitav) [reply packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:534]: Is it crudUploadThrottleMs? https://powersync-ja.github.io/powersync-react-native-sdk/common-sdk/interfaces/PowerSyncDatabaseOptions#cruduploadthrottlems

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts [hunk 10]
```diff
@@ -114,14 +116,25 @@ export abstract class AbstractPowerSyncDatabase extends BaseObserver<PowerSyncDB
...
+   * Schema used for the local database.
+   */
   get schema() {
     return this._schema;
   }
 
+  /**
+   * The underlying database.
+   * 
+   * For the most part, behavior is the same whether querying on the underlying database, or on [AbstractPowerSyncDatabase]. The main difference is in update notifications: the underlying database reports updates to the underlying tables, while AbstractPowerSyncDatabase reports updates to the higher-level views.
```
Comment (benitav) [inline packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:129]: Is this applicable here? Copied from Flutter.

File: packages/powersync-sdk-common/src/client/sync/bucket/CrudBatch.ts (hunk 6 above)
Comment (stevensJourney) [reply packages/powersync-sdk-common/src/client/sync/bucket/CrudBatch.ts:4]:
```suggestion
A batch of client-side changes.
```

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts (hunk 1 above)
Comment (stevensJourney) [reply packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:495]: This setting isn't currently represented in the common SDK. This is usually platform specific and should be handled by the database open factories. 

Web doesn't support multiple connections.

React native should implement this as an option in `RNQSPowerSyncDatabaseOpenFactory` - although that currently seems to be omitted from the configurable options.

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts (hunk 3 above)
Comment (stevensJourney) [reply packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:536]: This should be under `SQLWatchOptions::throttleMs`

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts (hunk 5 above)
Comment (stevensJourney) [reply packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:626]: This is just a private helper method. Not to be used by devs

File: packages/powersync-sdk-common/src/client/sync/bucket/CrudTransaction.ts [hunk 11]
```diff
@@ -1,10 +1,22 @@
...
+    /**
+     * List of client-side changes.
+     */
     public crud: CrudEntry[],
+    /**
+     * Call to remove the changes from the local queue, once successfully uploaded.
+     */
     public complete: (checkpoint?: string) => Promise<void>,
+    /**
+     * Unique transaction id.
```
Comment (stevensJourney) [inline packages/powersync-sdk-common/src/client/sync/bucket/CrudTransaction.ts:18]:
```suggestion
     * If null, this contains a list of changes recorded without an explicit transaction associated.
```

File: packages/powersync-sdk-common/src/db/crud/SyncStatus.ts (hunk 9 above)
Comment (stevensJourney) [reply packages/powersync-sdk-common/src/db/crud/SyncStatus.ts:25]:
```suggestion
   *  Time that a last sync has fully completed, if any.
   *  Currently this is reset to null after a restart.
```

File: packages/powersync-sdk-common/src/db/crud/SyncStatus.ts [hunk 12]
```diff
@@ -14,14 +14,23 @@ export type SyncStatusOptions = {
...
+   */
   get lastSyncedAt() {
     return this.options.lastSyncedAt;
   }
 
+  /**
+   * TODO
```
[the remark this answers is not in the document]
Comment (stevensJourney) [reply packages/powersync-sdk-common/src/db/crud/SyncStatus.ts:32]:
```
  /// true if actively downloading changes.
  ///
  /// This is only true when [connected] is also true.
  final bool downloading;

  /// true if uploading changes
  final bool uploading;
```

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts [hunk 13]
```diff
@@ -114,14 +116,25 @@ export abstract class AbstractPowerSyncDatabase extends BaseObserver<PowerSyncDB
...
+   * Schema used for the local database.
+   */
   get schema() {
     return this._schema;
   }
 
+  /**
+   * The underlying database.
+   * 
+   * For the most part, behavior is the same whether querying on the underlying database, or on {@link AbstractPowerSyncDatabase}. The main difference is in update notifications: the underlying database reports updates to the underlying tables, while AbstractPowerSyncDatabase reports updates to the higher-level views.
```
Comment (stevensJourney) [inline packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:129]:
```suggestion
   * For the most part, behavior is the same whether querying on the underlying database, or on {@link AbstractPowerSyncDatabase}.
```

File: packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts [hunk 14]
```diff
@@ -466,6 +490,12 @@ export abstract class AbstractPowerSyncDatabase extends BaseObserver<PowerSyncDB
     });
   }
 
+  /**
+   * Open a read-only transaction.
+   * TODO: Up to maxReaders read transactions can run concurrently. After that, read transactions are queued.
```
Comment (benitav) [inline packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts:495]:
```suggestion
```

**Review by kobiebotha — APPROVED** [review]
